src/Raven.Server/Documents/Handlers/AI/Agents/ConversationHandler.cs

```
@@ -354,103 +365,219 @@ public AiConnectionString GetAiConnectionString()
     {
         if (toolCalls == null || toolCalls.Count == 0)
             return;
 
         DynamicJsonArray reqs = [];
-        List<string> toolCallsIds = [];
-        var queryUrl = $"/databases/{database.Name}/queries";
+        List<AiToolCall> activeToolCalls = [];
         foreach (var call in toolCalls)
         {
-            var q = _configuration.FindQuery(call.Name);
-            if (q is null)
-                continue;
-
-            toolCallsIds.Add(call.Id);
-            reqs.Add(new DynamicJsonValue
+            switch (FindToolFrom(_configuration, call.Name))
             {
-                [nameof(GetRequest.Url)] = queryUrl,
-                [nameof(GetRequest.Query)] = null,
-                [nameof(GetRequest.Method)] = "POST",
-                [nameof(GetRequest.Content)] = new DynamicJsonValue
-                {
-                    [nameof(IndexQuery.Query)] = q.Query,
-                    [nameof(IndexQuery.QueryParameters)] = CreateParameters(context, call, _document.Parameters)
-                }
-            });
+                case AiAgentToolQuery q:
+                    activeToolCalls.Add(call);
+                    BuildQueryRequest(context, _document, reqs, q, call);
+                    break;
+                case AiAgentToolSubAgent agent:
+                    try
+                    {
+                        BuildAgentRequest(context, _document, call, agent, reqs);
+                    }
+                    catch (Exception e)
+                    {
+                        _document.AddMessage(context, context.ReadObject(
+                            new DynamicJsonValue
+                            {
+                                ["tool_call_id"] = call.Id,
+                                ["role"] = "tool",
+                                ["content"] = "Failure calling " + call.Id + ", because: " + e.Message
```

You are:
* hiding what agent was called (`call.Id` is a random value, you need the `call.Name`, the tool id is already passed in the `tool_call_id`)
* Hiding the actual error because it may be in a nested exception

---

Given that we send this to the model, and tokens count, we don't need all the stack trace, but the full list of exceptions + messages recursively matters

---

Done